assets/js/modules/analytics-4/datastore/conversion-reporting.js

```diff
@@ -275,43 +276,114 @@ export const selectors = {
 	 */
 	haveConversionEventsForTailoredMetrics: createRegistrySelector(
 		( select ) => ( state, useNewEvents ) => {
```

Comment:
I refactored this for simplicity, after changing the setting type this can be checked in more efficient way, using the detected events as more reliable argument